aeppl/loglik.py

```diff
@@ -87,30 +91,68 @@ def loglik(
 
     rng, size, dtype, *dist_params = rv_node.inputs
 
-    # Here, we plug the actual random variable into the log-likelihood graph,
-    # because we want a log-likelihood graph that only contains
-    # random variables.  This is important, because a random variable's
-    # parameters can contain random variables themselves.
-    # Ultimately, with a graph containing only random variables and
-    # "deterministics", we can simply replace all the random variables with
-    # their value variables and be done.
-
-    # tmp_rv_values = rv_values.copy()
-    # tmp_rv_values[rv_var] = rv_var
-
-    logpdf_var = _logpdf(rv_node.op, rv_value_var, *dist_params, **kwargs)
-
     # Replace random variables with their value variables
     replacements = rv_values.copy()
     replacements.update({rv_var: rv_value, rv_value_var: rv_value})
 
-    (logpdf_var,), _ = rvs_to_value_vars(
-        (logpdf_var,),
-        initial_replacements=replacements,
-    )
+    from collections import deque
+
+    # TODO: Walk the graph from its inputs to its outputs and construct the
+    # log
+    # TODO: Accept more than one graph.
+    # logp_fn = FunctionGraph(outputs=[rv_var], clone=False)
+
+    q = deque(io_toposort(graph_inputs((rv_var,)), (rv_var,)))
+
+    logpdf_var = None
```

Comment:
`logpdf_var = 0` ? No need for the if statement below, then.I assume a `0 +` is optimized away automatically...

Reply:
Yes, it should be removed during canonicalization, but then it's just a matter of when and where you want to do that work.  Since it doesn't really cost us anything to do that here, I would rather avoid introducing terms that would necessarily be removed.